### PR TITLE
CLOUDP-278494: Fix conversion/comparison of shared clusters

### PR DESCRIPTION
**Pull request by helderjs**
### All Submissions:

* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes-template.md if your changes should be included in the release notes for the next release.

**Note**: This change doesn't allow updating shared cluster tier (which is not allowed via public API) but process the change and  reflects the error on status.
image

### CODE REVIEW

File: internal/translation/deployment/conversion.go [hunk 1]
```diff
@@ -474,8 +474,8 @@ func labelsFromAtlas(cLabels []admin.ComponentLabel) []common.LabelSpec {
 }
 
 func replicationSpecFromAtlas(replicationSpecs []admin.ReplicationSpec) []*akov2.AdvancedReplicationSpec {
-	hSpecOrDefault := func(spec admin.HardwareSpec) *akov2.Specs {
-		if spec.GetNodeCount() == 0 {
+	hSpecOrDefault := func(spec admin.HardwareSpec, providerName string) *akov2.Specs {
+		if spec.GetNodeCount() == 0 && providerName != string(provider.ProviderTenant) {
```
Comment (josvazg) [inline internal/translation/deployment/conversion.go:478]: Why was this also necessary?

Comment (helderjs) [reply internal/translation/deployment/conversion.go:478]: during comparison, we were turning node spec with nodeCount 0 to nil. As the shared cluster has only instance size defined (nodeCount will be always 3 and the API omits the value :man_shrugging: ), we need to consider this when preparing it to compare.

Comment (josvazg) [reply internal/translation/deployment/conversion.go:478]: sorry, but what does the provider name have to do with the node count or instance size? I am missing that detail

Comment (helderjs) [reply internal/translation/deployment/conversion.go:478]: we identify shared cluster by provider name (TENANT)

**Review by josvazg — APPROVED** [review]

**Review by s-urbaniak — APPROVED** [review]

**Pull request merged**